clang/lib/CConv/ConstraintsGraph.cpp

```diff
@@ -50,27 +57,40 @@ void ConstraintsGraph::addConstraint(Geq *C, Constraints &CS) {
   }
   auto V1 = addVertex(A1);
   auto V2 = addVertex(A2);
+  add_edge(V2, V1, CG);
+}
 
-  EdgeType edgeType;
-  if (C->constraintIsChecked()) {
-    edgeType = Checked;
-  } else {
-    edgeType = Ptype;
-  }
 
-  add_edge(V2, V1, edgeType, CG);
+void GraphVizOutputGraph::mergeConstraintGraph(ConstraintsGraph Graph,
+                                               EdgeType EdgeType) {
+  for (auto E : Graph.getAllEdges()) {
+    auto S = addVertex(E.first);
+    auto T = addVertex(E.second);
+    add_edge(S, T, EdgeType, CG);
+  }
 }
 
-void ConstraintsGraph::dumpCGDot(const std::string& GraphDotFile) {
+void GraphVizOutputGraph::dumpCGDot(const std::string& GraphDotFile) {
    std::ofstream DotFile;
    DotFile.open(GraphDotFile);
    write_graphviz(DotFile, CG,
-       [&] (std::ostream &out, unsigned v) {
-         out << "[label=\"" << CG[v]->getStr() << "\"]";
-       },
-       [&] (std::ostream &out, boost::detail::edge_desc_impl<boost::bidirectional_tag, long unsigned int> e)  {
-         std::string color = EdgeTypeColors[CG[e]];
-         out << "[color=\"" << color << "\"]";
-       });
+     [&] (std::ostream &out, unsigned v) {
+       out << "[label=\"" << CG[v]->getStr() << "\"]";
+     },
+     [&] (std::ostream &out,
+              boost::detail::edge_desc_impl<boost::bidirectional_tag,
+                                            long unsigned int> e)  {
+       std::string Color = EdgeTypeColors[CG[e]];
+       out << "[color=\"" << Color << "\"]";
+     });
    DotFile.close();
+}
+
+void GraphVizOutputGraph::dumpConstraintGraphs(const std::string &GraphDotFile,
+                                                     ConstraintsGraph Chk,
```

Comment:
I think we want to make these arguments references, so they are not copied. (I mean `Chk` and `Pty`.)